### DISCUSSION TITLE
Integrate transformations into app/api

### DISCUSSION THREAD

**Discussion by Dominic-Marcelino** [directus/directus#12010] [Feature Requests]
Currently directus doesn’t know about the generated image transformations. 

In order to clean up files, manage transformations and organise all assets I think directus should store informations about the generated transformations (linked to the main file entry) and add a hook for file transformations.  

Some use-cases:
* directly use a custom version in the app (e.g create a relation to the 1x1 version of the image)
* Track all files (e.g to calculate the size of all files)
* manage transformations (e.g delete old ones that are no longer needed)

**Comment by benhaynes** [comment]
I agree. I think the "on the fly" aspect of the generated is ideal... but we also want to allow better tracking and management of these transformations. I defer to @rijkvanzanten on how we can track all of these without having to write them all to the database... unless that's what we want to do. :)

**Reply by Dominic-Marcelino** [reply]
If you don't want to track them in the db I think a hook would be a great MVP to fit into 80/20. This would allow everyone to create a custom logic to fits their need 🤔

Same goes for some kind of "before-upload" filter and "file.delete" hook

**Comment by Dominic-Marcelino** [comment]
Just noticed that `files.delete` does [exist](https://github.com/directus/directus/pull/5334/files#diff-989e6837090c2d3ad803a8b7c8bd70c3b1d7706ce90732839b0d61eb1dd82dcc) - it's just missing in the documentation.

Also I'm curious what has been the reason to disable to disable the `file.update` and `file.create` events? #5334 
I think they can be used for different things:

* `files.create`: filter / prevent uploads within hooks before the file is actually uploaded 
* `files.upload`: hook actions once the file is completely uploaded. E.g automatically create transformations

**Reply by benhaynes** [reply]
CC @erondpowell  😉

**Reply by Dominic-Marcelino** [reply]
resolved by #12094

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
